# Changes made to the Historical-transaction-data.csv
- Removed rows with less than 1 quantity_sold value:  8351
- No of item_description values generated:  1994
- No of invoice_ids generated:  4228

## Above changes are saved to a new csv file Historical-transaction-cleaned-data.csv

In [6]:
import pandas as pd

historical_transactions = pd.read_csv("data/Historical-transaction-cleaned-data.csv")

In [7]:
historical_transactions.head()

item_description          transaction_date  invoice_id customer_id   
0     ORANGE BARLEY 1.5L  2021-12-11T00:00:00.000Z       147.0        BGXA  \
1       GINGER BEER 1.5L  2021-10-17T00:00:00.000Z       371.0        IA25   
2        TONIC PET 500ML  2021-12-13T00:00:00.000Z       484.0        VN7V   
3          CREAM SODA 1L  2021-12-13T00:00:00.000Z       484.0        VN7V   
4  STRAWBERRY MILK 180ML  2021-10-23T00:00:00.000Z      1310.0        7S00   

   shop_id  item_price  quantity_sold  
0  SHOP008         220              2  
1  SHOP112         220              2  
2  SHOP008         160              2  
3  SHOP008         150              2  
4  SHOP112         210              5

In [8]:
historical_transactions.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 465623 entries, 0 to 465622
Data columns (total 7 columns):
 #   Column            Non-Null Count   Dtype  
---  ------            --------------   -----  
 0   item_description  465623 non-null  object 
 1   transaction_date  465623 non-null  object 
 2   invoice_id        465623 non-null  float64
 3   customer_id       465623 non-null  object 
 4   shop_id           465623 non-null  object 
 5   item_price        465623 non-null  int64  
 6   quantity_sold     465623 non-null  int64  
dtypes: float64(1), int64(2), object(4)
memory usage: 24.9+ MB


In [9]:
historical_transactions.describe()

invoice_id     item_price  quantity_sold
count  4.656230e+05  465623.000000  465623.000000
mean   2.080991e+06     207.493713       1.960066
std    1.520811e+06     167.877778       1.628437
min    1.700000e+01      35.000000       1.000000
25%    1.038512e+06     100.000000       1.000000
50%    2.046136e+06     200.000000       2.000000
75%    3.070249e+06     220.000000       2.000000
max    8.335982e+06   17400.000000     101.000000

In [10]:
print("\n* Historical-transaction-cleaned-data.csv SUMMARY *\n")
# Column names
print(f"{', '.join(historical_transactions.columns.tolist())} are the column names")
# No of rows
print(f"{len(historical_transactions)} total rows")

# Unique shop ids
shop_ids = historical_transactions["shop_id"].unique()
print(f"{len(shop_ids)} unique shop_id")

# Unique items
item_ids = historical_transactions["item_description"].unique()
print(f"{len(item_ids)} unique item_description")

# Unique transaction date
transaction_dates = historical_transactions["transaction_date"].unique()
print(f"{len(transaction_dates)} unique transaction_date")

# Unique item price
item_prices = historical_transactions["item_price"].unique()
print(f"{len(item_prices)} unique item_price")

# Unique quantity sold
quantities_sold = historical_transactions["quantity_sold"].unique()
print(f"{len(quantities_sold)} unique quantity_sold")


* Historical-transaction-cleaned-data.csv SUMMARY *

item_description, transaction_date, invoice_id, customer_id, shop_id, item_price, quantity_sold are the column names
465623 total rows
124 unique shop_id
2031 unique item_description
62 unique transaction_date
209 unique item_price
67 unique quantity_sold


## The Historical-transaction-cleaned-data.csv summary indicates that the cleaning process was successful. Here is the updated summary:
- 473,962 total rows
- 124 unique shop_id
- 2,031 unique item_description (Now includes the generated product names for rows with missing item_description)
- 62 unique transaction_date
- 209 unique item_price
- 68 unique quantity_sold (No longer includes -1 as a value)

## This cleaned dataset should be more suitable for analysis and generating features to make better predictions.